controllers/etcdpeer_controller.go

```diff
@@ -270,6 +291,82 @@ func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
+	if peer.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Add the PVC cleanup finalizer when we first see the peer
+		if !hasPvcDeletionFinalizer(peer) {
+			log.V(2).Info("Adding PVC cleanup finalizer")
+			updated := peer.DeepCopy()
+			updated.ObjectMeta.Finalizers = append(
+				updated.ObjectMeta.Finalizers,
+				pvcCleanupFinalizer,
+			)
+			err := r.Patch(ctx, updated, client.MergeFrom(&peer))
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to add PVC cleanup finalizer: %w", err)
+			}
+			log.V(2).Info("Added PVC cleanup finalizer")
+			return ctrl.Result{}, nil
+		}
+	} else {
+		// Peer has been deleted. Delete the PVC if the peer has been marked Decommissioned.
+		// And remove the PVC cleanup finalizer so the the peer can be garbage collected.
+		if hasPvcDeletionFinalizer(peer) {
+			if peer.Spec.Decommissioned {
```

Comment:
Looking at how this bool field is used, I'm not convinced it is necessary to have at all. The presence of the finalizer + the fact that this peer has been deleted should be enough to determine that the PVC should be deleted, if we only ever add the finalizer upon scale down. i.e. scale down process becomes:

1) call MemberRemove on the member
2) add finalizer to the EtcdPeer
3) delete the EtcdPeer resource